frontend/packages/kubevirt-plugin/src/components/cdi-upload-provider/upload-pvc-form/upload-pvc-form.tsx

```diff
@@ -0,0 +1,375 @@
+        Upload Data
+      </label>
+      <div className="form-group">
+        <FileUpload
+          id="file-upload"
+          value={fileValue}
+          filename={fileName}
+          onChange={handleFileChange}
+          hideDefaultPreview
+          isRequired
+        />
+      </div>
+      <div className="form-group">
+        <StorageClassDropdown
+          onChange={handleStorageClass}
+          id="storageclass-dropdown"
+          describedBy="storageclass-dropdown-help"
+          required={false}
+          name="storageClass"
+          filter={onlyPvcSCs}
+        />
+      </div>
+      <label className="control-label co-required" htmlFor="pvc-name">
+        Persistent Volume Claim Name
+      </label>
+      <div className="form-group">
+        <input
+          className="pf-c-form-control"
+          type="text"
+          onChange={handlePvcName}
+          placeholder="my-storage-claim"
+          aria-describedby="pvc-name-help"
+          id="pvc-name"
+          name="pvcName"
+          required
+        />
+        <p className="help-block" id="pvc-name-help">
+          A unique name for the storage claim within the project.
+        </p>
+      </div>
+      <label className="control-label co-required" htmlFor="access-mode">
+        Access Mode
+      </label>
+      <div className="form-group">
+        {accessModeRadios.map((radio) => {
+          let radioObj = null;
+          const disabled = !allowedAccessModes.includes(radio.value);
+
+          allowedAccessModes.forEach((mode) => {
+            const checked = !disabled ? radio.value === accessMode : radio.value === mode;
+            radioObj = (
+              <RadioInput
+                {...radio}
+                key={radio.value}
+                onChange={handleAccessMode}
+                inline
+                disabled={disabled}
+                checked={checked}
+                aria-describedby="access-mode-help"
+                name="accessMode"
+              />
+            );
+          });
+
+          return radioObj;
+        })}
+        <p className="help-block" id="access-mode-help">
+          {accessModeHelp}
+        </p>
+      </div>
+      <label className="control-label co-required" htmlFor="request-size-input">
+        Size
+      </label>
+      <RequestSizeInput
+        name="requestSize"
+        required
+        onChange={handleRequestSizeInputChange}
+        defaultRequestSizeUnit={requestSizeUnit}
+        defaultRequestSizeValue={requestSizeValue}
+        dropdownUnits={dropdownUnits}
+        describedBy="request-size-help"
+        inputID="request-size-input"
+      />
+      <p className="help-block" id="request-size-help">
+        Desired storage capacity.
+      </p>
+    </div>
+  );
+};
+
+export const UploadPVCPage: React.FC<UploadPVCPageProps> = (props) => {
+  const [isSubmitting, setIsSubmitting] = React.useState(false);
+  const [fileValue, setFileValue] = React.useState(null);
```

Comment:
this should probably be `string | File` right ?

```suggestion
  const [fileValue, setFileValue] = React.useState<string | File>();
```